build.sh

```diff
@@ -16,27 +14,14 @@ esac
 
 BUILDDIR=build/${BUILDTYPE}
 
-if ! hash meson 2>/dev/null && [ -x ${HOME}/.local/bin/meson ]
-then
-  export PATH=${PATH}:${HOME}/.local/bin
-fi
-
-if [ -f ${BUILDDIR}/build.ninja ]
-then
-  meson configure ${BUILDDIR} -Dbuildtype=${BUILDTYPE} -Dprefix=${INSTALL_PREFIX:-/usr/local} "$@"
-else
-  meson ${BUILDDIR} --buildtype ${BUILDTYPE} --prefix ${INSTALL_PREFIX:-/usr/local} "$@"
-fi
-
-cd ${BUILDDIR}
-
-NINJA=$(awk '/ninja/ {ninja=$4} END {print ninja}' meson-logs/meson-log.txt)
+MESON=$(PATH="${PATH}:${HOME}/.local/bin" command -v meson)
 
-if [ -n "${INSTALL_PREFIX}" ]
+if [ -f "${BUILDDIR}/build.ninja" ]
 then
-  ${NINJA} install
+  "${MESON}" configure "${BUILDDIR}" -Dbuildtype="${BUILDTYPE}" -Dprefix="${INSTALL_PREFIX:-/usr/local}" "$@"
 else
-  ${NINJA}
+  "${MESON}" "${BUILDDIR}" --buildtype "${BUILDTYPE}" --prefix "${INSTALL_PREFIX:-/usr/local}" "$@"
 fi
 
-popd
+"${MESON}" compile -C "${BUILDDIR}"
```

Comment:
Can you think of a simple way to check meson version? If not then I suppose we will have to increase the minimum version in `meson.build` to 0.55.

Reply:
According to https://mesonbuild.com/Commands.html#compile, 0.54 is enough, so I've bumped the requirement.

Reply:
`meson --version`? :)

Doing natural version sorting from a shell script is a bit fiddly, but if you assume GNU coreutils then pipe both version numbers to `sort -V` and then check whether the first (older version) line is the current version of Meson or the minimum one you need.

Reply:
Ubuntu 20.04 ships 0.53 it seems (18.04 ships 0.45 so not supported anymore). Should we really support these rather old versions if `pip install meson` does the trick on all those systems?

Reply:
I have asked on discord whether this would cause an issue for anyone there.